gifts_payments/models/gifts_payments.py

```diff
@@ -0,0 +1,88 @@
+from odoo import models, fields, api, exceptions
+import re
+
+
+class GiftsPayments(models.TransientModel):
+    _name = 'gifts.payments'
+
+    gifts_ids_text = fields.Text('Gifts IDs')
+
+    @api.multi
+    def do_gifts_search(self):
+        return {
+            'view_type': 'form',
+            'view_mode': 'form',
+            'res_model': 'gifts.results',
+            'type': 'ir.actions.act_window',
+            'target': 'new',
+            'context': str({'gifts_list': self.gifts_ids_text,
```

Comment:
context should not be placed in string. You should do this instead to pass the current context with your additions :
```
'context': self.with_context(gifts_list=self.gifts_ids_text, move_id=self.move_id.id).env.context
```
But even better, you should add those two fields in your 'gifts.results' class and pass them directly here instead of the context. To do so you could then do in the code:
```
results = self.env['gifts.results'].create({'gifts_list': self.gifts_ids_text, 'move_id': self.move_id.id})
return {
....
   'res_id': results.id
}
```
This will open your new record